src/tiled/tilesetdock.cpp

```diff
@@ -758,20 +758,31 @@ void TilesetDock::exportTileset()
     if (!tileset)
         return;
 
+    Preferences *prefs = Preferences::instance();
+
     const QLatin1String extension(".tsx");
-    QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
-    suggestedFileName += QLatin1Char('/');
+    QLatin1String empty(".");
+    QString suggestedFileName = prefs->lastPath(Preferences::ExternalTileset);
+
+    for(int i = (suggestedFileName.length() - 1); suggestedFileName.at(i) != QLatin1Char('/')
+        && suggestedFileName != empty; i--)
+        suggestedFileName.remove(i, 1);
```

Comment:
The for loop is required because when you suggest a path using the lastPath it will also include the file name of the last path inside the string. The foor loop loops from the end until it reaches a / to remove that and then add the Tileset's name to it.

What do you mean by that? How do you want me to change it? I don't really see an easier way to do it.